src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -124,12 +120,28 @@ object ScalafixPlugin extends AutoPlugin {
   )
 
   lazy val stdoutLogger = Compat.ConsoleLogger(System.out)
-  private var workingDirectory = file("").getAbsoluteFile.toPath
-  private var scalafixInterface: () => ScalafixInterface =
-    () => throw new UninitializedError
+
+  private val scalafixInterface: Def.Initialize[() => ScalafixInterface] =
```

Comment:
this is effectively like a `Setting`, so the function is created at sbt load time, but evaluated at task execution time, and since it's backed by a `LazyValue`, it's memoized.